src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -490,8 +515,33 @@ internal bool TryGetValue(int index, out DateTimeOffset value)
             ReadOnlySpan<byte> data = _utf8Json.Span;
             ReadOnlySpan<byte> segment = data.Slice(row.Location, row.SizeOrLength);
 
-            if (JsonHelpers.TryParseAsISO(segment, out DateTimeOffset tmp, out int bytesConsumed) &&
-                segment.Length == bytesConsumed)
+            DateTimeOffset tmp;
+            int bytesConsumed;
+
+            if (row.HasComplexChildren)
```

Comment:
Why do we need this check?

Reply:
This is "needs to be unescaped" :)

Reply:
Oh, lol. Got it. A comment would be helpful :)